### DISCUSSION TITLE
How to remove SSID1

### DISCUSSION THREAD

**Discussion by evansgl** [khoih-prog/ESPAsync_WiFiManager#106] [Q&A] [answered]
Any chance how to remove SSID1 from the configuration portal.
I only need one SSID to be configured.

**Comment by khoih-prog** [comment, marked as answer]
Check [**Single SSID instead of multiple SSIDs** #8](https://github.com/khoih-prog/ESPAsync_WiFiManager/issues/8)

> You can either
> 
> 1. Use the previous version [**ESPAsync_WiFiManager v1.0.11**](https://github.com/khoih-prog/ESPAsync_WiFiManager#releases-1011), which uses only single SSID without the MultiWiFi feature.
> 2. Just **ignore** the extra SSID or make them duplicated.
> 3. **Change in the src files of newer Releases v1.13.0+**

If there are **many more similar requests**, I'll possibly make single SSID **optional**, configurable in your code, certainly if having time.

**Reply by evansgl** [reply]
Thank you.I'll try to modify the code.  It would be nice to enable/disable with a "define" rule in the latest release.

**Reply by khoih-prog** [reply]
That's good. You can have a look and do something similar to what I actually did for [**ESP_WiFiManager_Lite** Library](https://github.com/khoih-prog/ESP_WiFiManager_Lite).  But certainly a little bit more complex for this library.

https://github.com/khoih-prog/ESP_WiFiManager_Lite/blob/bc8fb7ed5159e78f4cba35e07bf8cb18a7925320/examples/ESP_WiFi/defines.h#L98

```
#define REQUIRE_ONE_SET_SSID_PW               true    //false
```

https://github.com/khoih-prog/ESP_WiFiManager_Lite/blob/bc8fb7ed5159e78f4cba35e07bf8cb18a7925320/src/ESP_WiFiManager_Lite.h#L564-L567

https://github.com/khoih-prog/ESP_WiFiManager_Lite/blob/bc8fb7ed5159e78f4cba35e07bf8cb18a7925320/src/ESP_WiFiManager_Lite.h#L1323-L1341

```
// New from v1.3.0
#if !defined(REQUIRE_ONE_SET_SSID_PW)
  #define REQUIRE_ONE_SET_SSID_PW     false
#endif

...

      #if REQUIRE_ONE_SET_SSID_PW
      // If SSID ="blank" or NULL, or PWD length < 8 (as required by standard) => return false
      // Only need 1 set of valid SSID/PWD
      if (!( ( ( strncmp(ESP_WM_LITE_config.WiFi_Creds[0].wifi_ssid, WM_NO_CONFIG, strlen(WM_NO_CONFIG)) && strlen(ESP_WM_LITE_config.WiFi_Creds[0].wifi_ssid) >  0 )  &&
             (   strlen(ESP_WM_LITE_config.WiFi_Creds[0].wifi_pw) >= PASSWORD_MIN_LEN ) ) ||
             ( ( strncmp(ESP_WM_LITE_config.WiFi_Creds[1].wifi_ssid, WM_NO_CONFIG, strlen(WM_NO_CONFIG)) && strlen(ESP_WM_LITE_config.WiFi_Creds[1].wifi_ssid) >  0 )  &&
               ( strlen(ESP_WM_LITE_config.WiFi_Creds[1].wifi_pw) >= PASSWORD_MIN_LEN ) ) ))
      #else
      // If SSID ="blank" or NULL, or PWD length < 8 (as required by standard) => invalid set
      // Need both sets of valid SSID/PWD
      if ( !strncmp(ESP_WM_LITE_config.WiFi_Creds[0].wifi_ssid,   WM_NO_CONFIG, strlen(WM_NO_CONFIG) )  ||
           !strncmp(ESP_WM_LITE_config.WiFi_Creds[0].wifi_pw,     WM_NO_CONFIG, strlen(WM_NO_CONFIG) )  ||
           !strncmp(ESP_WM_LITE_config.WiFi_Creds[1].wifi_ssid,   WM_NO_CONFIG, strlen(WM_NO_CONFIG) )  ||
           !strncmp(ESP_WM_LITE_config.WiFi_Creds[1].wifi_pw,     WM_NO_CONFIG, strlen(WM_NO_CONFIG) )  ||
           ( strlen(ESP_WM_LITE_config.WiFi_Creds[0].wifi_ssid) == 0 ) || 
           ( strlen(ESP_WM_LITE_config.WiFi_Creds[1].wifi_ssid) == 0 ) ||
           ( strlen(ESP_WM_LITE_config.WiFi_Creds[0].wifi_pw)   < PASSWORD_MIN_LEN ) ||
           ( strlen(ESP_WM_LITE_config.WiFi_Creds[1].wifi_pw)   < PASSWORD_MIN_LEN ) )
      #endif     
```
